arangod/Agency/AgencyComm.cpp

```diff
@@ -1846,14 +1846,14 @@ bool AgencyComm::shouldInitializeStructure() {
       // Sanity
       if (result.slice().isArray() && result.slice().length() == 1) {
 
-        // No plan entry? Should initialise
-        if (result.slice()[0] == VPackSlice::emptyObjectSlice()) {
+        // No plan entry? Should initialize
+        if (result.slice()[0].isObject() && result.slice()[0].length() == 0) {
```

Comment:
I think there is `isEmptyObject()`

Reply:
@maierlars : there is.
However, its implementation currently is:
``` 
   // tests whether the Slice is an empty object
   constexpr bool isEmptyObject() const noexcept { return head() == 0x0a; }
```

Reply:
We should probably adjust this as well to make it more "safe".

Reply:
yes, please :+1: